### PR TITLE
Check for right-hand operand of zero in lrem and irem

### PR DESCRIPTION
**Pull request by hzongaro**
If the following program
 
```
public class Bug {
    public static final int sub(int i) {
        int[] arr = {1,2,3,4,5,6,7,8};
        long l = 0;
        return arr[-((int) (i % l))];
    }

    public static final void main(String[] args) {
        sub(1);
    }
}
```

is run with `java -Xjit:count=0 Bug`, the following assertion failure results:

```
Assertion failed at /root/hostdir/defects/issue15730/openj9-openjdk-jdk17/omr/compiler/optimizer/VPHandlers.cpp:8171: low <= high
VMState: 0x000514ff
        Node 0x7f87257752c0 [l2i]: reversed child constraint bounds
```

The problem is that the `constrainIrem` and `constrainLrem` Value Propagation handlers guard against a right-hand operand of zero if the left-hand operand is a constant, but not if the left-hand operand is a non-constant.  In the latter case, a constraint in the range [1,-1] results.  Notice that the upper bound of the range is less than the lower bound.

This change adds an additional check for a right-hand operand equal to zero if the left-hand operand is non-constant, avoiding creating an erroneous constraint for the result.

### CODE REVIEW

**Comment by hzongaro** [comment]
Devin @jdmpapin, may I ask you to review this fix?

**Comment by hzongaro** [comment]
Daryl @0xdaryl, may I ask you to review as well?

File: compiler/optimizer/VPHandlers.cpp [hunk 1]
```diff
@@ -8142,6 +8152,11 @@ static TR::Node *constrainNarrowIntValue(OMR::ValuePropagation *vp, TR::Node *no
    const int64_t min = -signBit;
    const int64_t max = signBit - 1;
 
+if (vp->trace())
+{
+traceMsg(vp->comp(), "In constrainNarrowIntValue:  resultSizeBits == %d; signBit == %d; padding64 == %d; min == %d; max -- %d\n", resultSizeBits, signBit, padding64, min, max);
```
Comment (jdmpapin) [inline compiler/optimizer/VPHandlers.cpp:8157]: This (and the below) tracing seems to have been unintentionally left here (e.g. it isn't indented)

**Review by jdmpapin — COMMENTED** [review]
Actual fix looks reasonable

File: compiler/optimizer/VPHandlers.cpp (hunk 1 above)
Comment (hzongaro) [reply compiler/optimizer/VPHandlers.cpp]: Sigh!  I'm sorry about that.  I've fixed it and force pushed the change again under commit 43ba52729a70112519803f92467cd844cc313c5b.

**Comment by hzongaro** [comment]
Devin @jdmpapin, thanks for the review.  I've removed the debugging trace messages that I had inadvertently left behind.  It's ready for another review.

**Review by jdmpapin — APPROVED** [review]

**Comment by jdmpapin** [comment]
Jenkins build all

**Comment by hzongaro** [comment]
@jdmpapin Devin, FYI - I removed @0xdaryl as a reviewer for this simple fix.  I had forgotten that you are a committer for OMR.

**Pull request merged**